### DISCUSSION TITLE
Facing failure on vpi3 test run : Below testcase skipping and failing, while running on the work-in-progress branch for the JetPack 6 Developer Preview

### DISCUSSION THREAD

**Discussion by PugalendhiMurugan** [OE4T/meta-tegra-community#111] [Q&A] [answered]
Failed Log:
root@nvidia-jetson-agx-orin-devkit:/# run-vpi3-tests
[278714.153621] nvgpu: 17000000.gpu             railgate_enable_store:365  [INFO]  railgate is disabled.
=== BEGIN: convolve_2d ===
Running 01_convolve_2d - Backend is cpu
=== PASS:  convolve_2d ===
=== BEGIN: convolve_2d ===
Running 01_convolve_2d - Backend is cuda
=== PASS:  convolve_2d ===
=== BEGIN: stereo_disparity ===
Running 02_stereo_disparity - Backend is cpu
Backend 'cpu' not recognized, it must be either cuda, ofa or ofa-pva-vic.
=== FAIL:  stereo_disparity ===
=== BEGIN: stereo_disparity ===
Running 02_stereo_disparity - Backend is cuda
=== PASS:  stereo_disparity ===
=== BEGIN: stereo_disparity ===
Running 02_stereo_disparity - Backend is ofa
=== PASS:  stereo_disparity ===
=== BEGIN: harris_corners ===
Running 03_harris_corners - Backend is cpu

184 keypoints found
=== PASS:  harris_corners ===
=== BEGIN: harris_corners ===
Running 03_harris_corners - Backend is cuda

184 keypoints found
=== PASS:  harris_corners ===
=== BEGIN: rescale ===
Running 04_rescale - Backend is cpu
=== PASS:  rescale ===
=== BEGIN: rescale ===
Running 04_rescale - Backend is cuda
=== PASS:  rescale ===
=== BEGIN: rescale ===
Running 04_rescale - Backend is vic
=== PASS:  rescale ===
=== BEGIN: benchmark ===
Running 05_benchmark - Backend is cpu
Input size: 1920 x 1080
Image format: VPI_IMAGE_FORMAT_U16
Algorithm: 5x5 Gaussian Filter
Approximated elapsed time per call: 0.561281 ms
=== PASS:  benchmark ===
=== BEGIN: benchmark ===
Running 05_benchmark - Backend is cuda
Input size: 1920 x 1080
Image format: VPI_IMAGE_FORMAT_U16
Algorithm: 5x5 Gaussian Filter
Approximated elapsed time per call: 0.197354 ms
=== PASS:  benchmark ===
=== BEGIN: klt_tracker ===
Skipping 06_klt_tracker
=== SKIP:  klt_tracker ===
=== BEGIN: fft ===
Running 07_fft - Backend is cpu
=== PASS:  fft ===
=== BEGIN: fft ===
Running 07_fft - Backend is cuda
=== PASS:  fft ===
=== BEGIN: tnr ===
Skipping 09_tnr
=== SKIP:  tnr ===
=== BEGIN: perspwarp ===
Skipping 10_perspwarp
=== SKIP:  perspwarp ===
=== BEGIN: fisheye ===
Running 11_fisheye
rms error: 55.685987
Fisheye coefficients: -0.073234 0.121251 -0.120230 0.038480
Camera matrix:
[440.956751 0.000000 659.822930; 0.000000 439.322114 451.685233; 0.000000 0.000000 1.000000]
=== PASS:  fisheye ===
=== BEGIN: optflow_lk ===
Skipping 12_optflow_lk
=== SKIP:  optflow_lk ===
=== BEGIN: optflow_dense ===
Skipping 13_optflow_dense
=== SKIP:  optflow_dense ===
=== BEGIN: background_subtractor ===
Skipping 14_background_subtractor
=== SKIP:  background_subtractor ===
=== BEGIN: image_view ===
Running 15_image_view - Backend is cpu
=== PASS:  image_view ===
=== BEGIN: template_matching ===
Running 17_template_matching - Backend is cpu
Provided coord of bounding box for the template is [200, 100] with w=100 and h=100
Template matching location coord is [100, 200]
=== PASS:  template_matching ===
=== BEGIN: template_matching ===
Running 17_template_matching - Backend is cuda
Provided coord of bounding box for the template is [200, 100] with w=100 and h=100
Template matching location coord is [100, 200]
=== PASS:  template_matching ===
=== BEGIN: orb_feature_detector ===
Running 18_orb_feature_detector - Backend is cpu
=== PASS:  orb_feature_detector ===
=== BEGIN: orb_feature_detector ===
Running 18_orb_feature_detector - Backend is cuda
=== PASS:  orb_feature_detector ===
Tests run:     26
Tests passed:  19
Tests skipped: 6
Tests failed:  1
*******************************
Failed: 
stereo_disparity

Skipped: 
optflow_lk
optflow_dense
background_subtractor
perspwarp
klt_tracker
tnr

**Comment by ichergui** [comment]
Thanks @PugalendhiMurugan  for reporting this issue. We need to update the backend used by the VPI application. 
`02_stereo_disparity` application doesn't support `cpu` backend.
I will fix that.

**Reply by PugalendhiMurugan** [reply]
Thank you for the info @ichergui

**Reply by PugalendhiMurugan** [reply]
Hi @ichergui ,
Other skipped cases are because of test files for those case not available on desired path, Does the script needs to be updated for those skip cases also.

**Comment by ichergui** [comment, marked as answer]
Hi @PugalendhiMurugan 
I tried the `run-vpi3-tests` script. It works as expected. 

Are you sure that you are using the latest commits `meta-tegra` and `tegra-tegra-community` ?

```
root@jetson-agx-orin-devkit:~# run-vpi3-tests
...
Processing frame 418
Processing frame 419
Processing frame 420
Processing frame 421
=== PASS:  background_subtractor ===
=== BEGIN: image_view ===
Running 15_image_view - Backend is cpu
=== PASS:  image_view ===
=== BEGIN: template_matching ===
Running 17_template_matching - Backend is cpu
Provided coord of bounding box for the template is [200, 100] with w=100 and h=100 
Template matching location coord is [100, 200] 
=== PASS:  template_matching ===
=== BEGIN: template_matching ===
Running 17_template_matching - Backend is cuda
Provided coord of bounding box for the template is [200, 100] with w=100 and h=100 
Template matching location coord is [100, 200] 
=== PASS:  template_matching ===
=== BEGIN: orb_feature_detector ===
Running 18_orb_feature_detector - Backend is cpu
=== PASS:  orb_feature_detector ===
=== BEGIN: orb_feature_detector ===
Running 18_orb_feature_detector - Backend is cuda
=== PASS:  orb_feature_detector ===
Tests run:     32
Tests passed:  32
Tests skipped: 
Tests failed:  0
root@jetson-agx-orin-devkit:~# 
```

You need to add the following line in the `conf/local.conf` to enable all the VPI sample applications:
```
LICENSE_FLAGS_ACCEPTED:append = " commercial_ffmpeg"
```

**Reply by PugalendhiMurugan** [reply]
Thank you @ichergui  I will give a try with above config.
